include/core/platform.h

```diff
@@ -208,7 +202,7 @@ extern int lf_cond_signal(lf_cond_t* cond);
  *
  * @return 0 on success, platform-specific error number otherwise.
  */
-extern int lf_cond_wait(lf_cond_t* cond, lf_mutex_t* mutex);
+extern int lf_cond_wait(lf_cond_t* cond);
```

Comment:
Removing the mutex argument from all these API functions seems dangerous to me.  It seems to force a design that has only a single global mutex, precluding any design with multiple mutexes. This could be a serious limitation. Is this really necessary?  The argument could be ignored for platforms that really do force just a single mutex.

Reply:
I don't think we force the use of a single mutex with this design; AFAIK the `lf_cond_t` type simply bundles a condition variable with a mutex (any mutex).